dask/dataframe/io.py

```diff
@@ -430,30 +430,60 @@ def to_hdf(df, path_or_buf, key, mode='a', append=False, complevel=0,
             warn("In order to preserve order between partitions "
                  "name_function must preserve the order of its input")
 
+    # if link is None, lets try our bast to avoid links
+    # if we're saving to multiple files we can assume we can skip links
+    # if we're appending, all must still link against first chunk
+    if link is None:
+        link = True
+        # We're always adding, no links needed
+        if mode == 'a' and append:
+            link = False
+        # we dont support link removals with non-unique name_functions, unless we're always appending
+        elif len(set(name_function(i) for i in range(df.npartitions))) != df.npartitions:
+            link = True
+        # if we're appending to files and have a file per partition, we don't need to link either
+        elif mode == 'a' and isinstance(path_or_buf, str) and path_or_buf.count('*'):
+            link = False
+        elif mode == 'w' and isinstance(path_or_buf, str) and path_or_buf.count('*'):
+            link = 'first'
+        # if we're appending to nodes and have a node per partition, we don't need to link either
+        elif append and key.count('*'):
+            link = False
+
     dsk = dict()
     i_name = name_function(0)
-    dsk[(name, 0)] = (_link, None,
-                      (apply, pd_to_hdf,
-                          (tuple, [(df._name, 0), fmt_obj(path_or_buf, i_name),
-                              key.replace('*', i_name)]),
-                          merge(kwargs,
-                            {'mode':  mode, 'format': 'table', 'append': append,
-                             'complevel': complevel, 'complib': complib,
-                             'fletcher32': fletcher32})))
+    task = (apply, pd_to_hdf,
+               (tuple, [(df._name, 0), fmt_obj(path_or_buf, i_name),
+                   key.replace('*', i_name)]),
```

Comment:
We can probably drop the delayed call to `tuple`.  The schedulers have evolved a bit since this was first written